### PR TITLE
Update RouteAggregator.js

### PR DESCRIPTION
**Pull request by SimonDegraeve**
`isRelayContainer` check for `hasFragment` function.

https://github.com/facebook/relay/blob/931df0de2666b63f251d090a9edfe1e6d0ac0e98/src/container/isRelayContainer.js

### CODE REVIEW

File: src/RouteAggregator.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ export default class RouteAggregator {
         const uniqueQueryName = this._getUniqueQueryName(query, queryName);
 
         relayRoute.queries[uniqueQueryName] =
-          () => query(component, routeParams);
+          () => typeof query === 'function' ? query(component, routeParams) : query;
```
Comment (taion) [inline src/RouteAggregator.js]: What's this for? Can queries ever not be functions?

**Comment by SimonDegraeve** [comment]
Indeed yes. https://github.com/facebook/relay/commit/077bb16f64dfd44a9d739689a1546dc66a37c34b

**Comment by taion** [comment]
Those queries all still look like functions to me.

**Comment by SimonDegraeve** [comment]
;-) ... my bad, already late here

File: src/RouteAggregator.js [hunk 2]
```diff
@@ -126,4 +126,8 @@ export default class RouteAggregator {
     const {component, queryName} = this._fragmentSpecs[fragmentName];
     return component.getFragment(queryName, variableMapping);
   }
+  
+  hasFragment(fragmentName) {
+    return !!this._fragmentSpecs[fragmentName];
```
Comment (taion) [inline src/RouteAggregator.js]: Let's use an explicit check for `undefined` here just to be on the safe side.

File: src/RouteAggregator.js (hunk 1 above)
Comment (taion) [inline src/RouteAggregator.js]: Okay - if queries are always functions, then we should get rid of this change.

Comment (SimonDegraeve) [inline src/RouteAggregator.js]: Done.

**Comment by taion** [comment]
Ah crud. This actually is going to require a more extensive re-write. If two routes use the same queries but define different fragments corresponding to those queries, then the current implementation will break when using the new shorthand.

Thanks a lot @devknoll :p

**Comment by SimonDegraeve** [comment]
Ok, can you merge this so at least the `RelayContainer PropType` doesn't fail. And maybe add a warning on the readme about the shorthand not working for now...

**Comment by taion** [comment]
I manually squashed and merged.

**Comment by SimonDegraeve** [comment]
Alright, thanks.

**Comment by devknoll** [comment]
Ahaha, sorry @taion :wink: 

> And maybe add a warning on the readme about the shorthand not working for now...

This seems like a decent solution for now.

**Comment by taion** [comment]
@devknoll

I don't need to! Almost have a fix ready. @josephsavona's thing at https://github.com/facebook/relay/pull/115#discussion_r37831400 means that I don't need to re-write my query proxying code at all. I just need to generate unique query names by route rather than by query in case the same query is used for multiple routes.

**Comment by devknoll** [comment]
:+1:

**Comment by taion** [comment]
(And this was actually just a bug anyway, the "not supporting using the same query for multiple routes" part.)

**Pull request closed without merging**